### ISSUE TITLE
no-unused-prop-types throwing error for destructured props

### ISSUE THREAD

**Issue by woodenconsulting** [yannickcr/eslint-plugin-react#2424]
# Sample Code

```javascript
function myStatelessComponent(props) {
  const { test: { foo, bar } } = props;
  return (
    <div>
      {foo} {bar}
    </div>
  );
}

myStatelessComponent.propTypes = {
  test: PropTypes.shape({
    foo: PropTypes.string.isRequired,
    bar: PropTypes.string.isRequired
  }).isRequired
};
```

# Issue 

The linter erroneously throws on the `test` prop since only its descendants are used: 
`'test' PropType is defined but prop is never used  react/no-unused-prop-types`.

The issue first appeared with the [7.15.0 release](https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.15.0) published about 11 hours ago from the time of this writing. It's likely due to [this commit](https://github.com/yannickcr/eslint-plugin-react/commit/66725bcf2234938b1a1e76c58b070dd6332523c0).

# Expected

The linter should not throw

**Comment by ljharb** [comment]
cc @golopot

**Comment by golopot** [comment]
I cannot reproduce this. I added the test case  in my fork branch and CI is passing.  https://github.com/golopot/eslint-plugin-react/tree/eeee
https://travis-ci.org/golopot/eslint-plugin-react/builds/592117332

I have tried `espree`, `babel-eslint`, `@typescript-eslint/parser`, `env: {mocha: true}`, and declaring a local variable named `test`.

**Comment by ljharb** [comment]
a PR with the test case would be appreciated.

@woodenconsulting can you provide any more info that would be helpful?

**Comment by woodenconsulting** [comment]
@golopot @ljharb I created repo which reproduces the issue: https://github.com/woodenconsulting/epr-2424

## Additional Notes

- Unlike the code I first provided, the issue is only prevalent when I [import the shape](https://github.com/woodenconsulting/epr-2424/blob/master/src/index.jsx#L3)
- The issue also goes away when I remove the `eslint-config-airbnb` plugin that we're using.

I haven't had a chance to dig any further; however, for additional context this error is happening across several private repos where pinning to the prior version of `eslint-plugin-react` solves the issue (for now).

**Comment by ljharb** [comment]
aha, that first bullet point seems like the trick.

**Comment by redbugz** [comment]
If it helps, this is stack trace that we get:
```
TypeError: Cannot read property 'substring' of undefined
Occurred while linting /home/travis/build/my-org/my-repo/packages/contact-card/src/index.js:29
    at Object.fix (/home/travis/build/my-org/my-repo/node_modules/eslint-plugin-react/lib/rules/jsx-curly-brace-presence.js:129:30)
    at normalizeFixes (/home/travis/build/my-org/my-repo/node_modules/eslint/lib/util/report-translator.js:176:28)
    at args (/home/travis/build/my-org/my-repo/node_modules/eslint/lib/util/report-translator.js:278:18)
    at Object.report (/home/travis/build/my-org/my-repo/node_modules/eslint/lib/linter.js:721:41)
    at reportUnnecessaryCurly (/home/travis/build/my-org/my-repo/node_modules/eslint-plugin-react/lib/rules/jsx-curly-brace-presence.js:117:15)
    at lintUnnecessaryCurly (/home/travis/build/my-org/my-repo/node_modules/eslint-plugin-react/lib/rules/jsx-curly-brace-presence.js:198:9)
    at JSXExpressionContainer (/home/travis/build/my-org/my-repo/node_modules/eslint-plugin-react/lib/rules/jsx-curly-brace-presence.js:278:11)
```

**Comment by w33ble** [comment]
Also seeing the `TypeError: Cannot read property 'substring' of undefined` error, in the following JSX:

```jsx
<Tooltip content={<div>Hello World</div>} placement={placement}>
  <Button color="primary" variant="contained">
    I have a tooltip ({placement})
  </Button>
</Tooltip>
```

The error is occurring on the first line there. Seems to be a new error as of v7.15.0.

**Comment by ljharb** [comment]
@w33ble @redbugz that error should be fixed in v7.15.1.

**Comment by woodenconsulting** [comment]
@golopot Your [fix](https://github.com/yannickcr/eslint-plugin-react/pull/2428) does indeed resolve the issue, thanks for the quick turn around! 💯

**Comment by w33ble** [comment]
Can confirm. Thanks for the fix and the notification. 🥇
